src/test/java/com/griddynamics/cloud/learning/service/ArticleServiceTest.java

```diff
@@ -0,0 +1,356 @@
+        final Pageable pageable = PageRequest.of(0, 3, Sort.by(Sort.Order.desc("date")));
+
+        final Article expectedArticle1 = Article.builder().id(1L).caption("first").content("first content").build();
+        final Article expectedArticle2 = Article.builder().id(2L).caption("second").content("second content").build();
+        final Article expectedArticle3 = Article.builder().id(3L).caption("third").content("third content").build();
+
+        final ArticleDto expectedArticleDto1 = ArticleDto.builder().id(1L).caption("first").build();
+        final ArticleDto expectedArticleDto2 = ArticleDto.builder().id(2L).caption("second").build();
+        final ArticleDto expectedArticleDto3 = ArticleDto.builder().id(3L).caption("third").build();
+
+        final PageImpl<Article> articlesPage = new PageImpl<>(List.of(expectedArticle1, expectedArticle2, expectedArticle3),
+                pageable, 50);
+
+        when(repository.findAll(any(Pageable.class))).thenReturn(articlesPage);
+        when(mapper.articleToArticleDto(expectedArticle1)).thenReturn(expectedArticleDto1);
+        when(mapper.articleToArticleDto(expectedArticle2)).thenReturn(expectedArticleDto2);
+        when(mapper.articleToArticleDto(expectedArticle3)).thenReturn(expectedArticleDto3);
+
+        //when
+        final Page<ArticleDto> articles = service.findArticles(request, pageable);
+
+        //then
+        ArgumentCaptor<Pageable> acPageable = ArgumentCaptor.forClass(Pageable.class);
+        verify(repository, times(1)).findAll(acPageable.capture());
+        assertEquals(pageable, acPageable.getValue());
+
+        assertEquals(50, articles.getTotalElements());
+        assertEquals(17, articles.getTotalPages());
+        assertEquals(3, articles.getContent().size());
+        assertEquals(new HashSet<>(articles.getContent()), Set.of(expectedArticleDto1, expectedArticleDto2, expectedArticleDto3));
+    }
+
+    @Test
+    public void shouldGetArticlesByPurchaseType() {
+        //given
+        final ArticleFilterRequest request = new ArticleFilterRequest(PurchaseType.FREE, "", "");
+        final Pageable pageable = PageRequest.of(4, 2, Sort.by(Sort.Order.desc("caption")));
+
+        final Article expectedArticle1 = Article.builder().id(1L).caption("first").content("first content").isFree(true).build();
+        final Article expectedArticle2 = Article.builder().id(2L).caption("second").content("second content").isFree(true).build();
+
+        final ArticleDto expectedArticleDto1 = ArticleDto.builder().id(1L).caption("first").isFree(true).build();
+        final ArticleDto expectedArticleDto2 = ArticleDto.builder().id(2L).caption("second").isFree(true).build();
+
+        final PageImpl<Article> articlesPage = new PageImpl<>(List.of(expectedArticle1, expectedArticle2),
+                pageable, 10);
+
+        when(repository.findAllByIsFree(anyBoolean(), any(Pageable.class))).thenReturn(articlesPage);
```

Comment:
Please follow your scenario, should be exactly TRUE instead of anyBoolean()

Please make sure that you other scenarios are also consistent